google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/FileConfirmationDialog.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gct.idea.appengine.cloud;
+
+import com.google.gct.idea.util.GctBundle;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.util.SystemInfo;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+/**
+ * A dialog warning for, e.g., overwriting existing app.yaml and Dockerfile configuration files
+ * when user selects a destination folder to auto-generate them.
+ */
+public class FileConfirmationDialog extends DialogWrapper {
```

Comment:
There is no direct option to limit the length. I could probably do it by subclassing Message.InputDialog or something, but then we're not solving our problem to get away from a new class. I see basically two options: 1) just use one of the Messages functions and forget about the length; 2) live it with the current FileConfirmationDialog implementation.